### ISSUE TITLE
Link errors when linking HDF5 against OpenMPI 4.0.0

### ISSUE THREAD

**Issue by dorier** [open-mpi/ompi#6116]
### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)

4.0.0 installed from the archive provided on the website.

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

OpenMPI was installed manually from the website-provided archive.

### Please describe the system on which you are running

* Operating system/version: Debian 9
* Computer hardware: x86_64
* Network type: Ethernet

-----------------------------

## Details of the problem

After I managed to install OpenMPI 4.0.0 (see issue https://github.com/open-mpi/ompi/issues/6115) I made spack aware of the installation by adding it to my `.spack/linux/packages.yaml`. I then tried to build HDF5 using spack, on top of this installation of OpenMPI:

```
spack install hdf5 ^openmpi
```

I'm getting those link errors:

```
==> Error: ProcessError: Command exited with status 2:
    'make' '-j2'

9 errors found in build log:
     4684      CCLD     libfilter_plugin1_dsets.la
     4685      CCLD     libfilter_plugin2_dsets.la
     4686      CCLD     libfilter_plugin3_dsets.la
     4687      CCLD     libfilter_plugin4_groups.la
     4688      CCLD     testhdf5
     4689      CCLD     cache
  >> 4690    ../src/.libs/libhdf5.so: undefined reference to `MPI_Type_extent'
  >> 4691    ../src/.libs/libhdf5.so: undefined reference to `OMPI_C_MPI_NULL_COPY_FN'
  >> 4692    collect2: error: ld returned 1 exit status
     4693    Makefile:1991: recipe for target 'testhdf5' failed
  >> 4694    make[1]: *** [testhdf5] Error 1
     4695    make[1]: *** Waiting for unfinished jobs....
  >> 4696    ../src/.libs/libhdf5.so: undefined reference to `MPI_Type_extent'
  >> 4697    ../src/.libs/libhdf5.so: undefined reference to `OMPI_C_MPI_NULL_COPY_FN'
  >> 4698    collect2: error: ld returned 1 exit status
     4699    Makefile:1647: recipe for target 'cache' failed
  >> 4700    make[1]: *** [cache] Error 1
     4701    make[1]: Leaving directory '/tmp/mdorier/spack-stage/spack-stage-2DLw0P/hdf5-1.10.4/test'
     4702    Makefile:654: recipe for target 'all-recursive' failed
  >> 4703    make: *** [all-recursive] Error 1
```

**Comment by dorier** [comment]
I realize that the link errors are caused by symbols that are deprecated so it should fall on HDF5 to be updated. I will report this to the HDF5 community instead, and close this issue.

**Comment by giuseppe2590** [comment]
Hello, I have the same problem. Do you have any solutions?

**Comment by dorier** [comment]
As mentioned above, this is an HDF5 problem. I reported it here: https://forum.hdfgroup.org/t/hdf5-does-not-compile-with-openmpi-4-0-0/5110/2
Apparently this should be fixed for the 1.10.5 release, which is not out yet, so we will have to wait a bit.

**Comment by edgargabriel** [comment]
as mentioned in the hdf5 ticket, if you configure and compile Open MPI v4.0.0 with the --enable-mpi1-compatibility flag, it actually works.
